and_scale_ros/scripts/ekew_i_driver.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+
+import serial
+
+from jsk_recognition_msgs.msg import WeightStamped
+import rospy
+
+
+class EkEwIDriver(object):
+
+    """Read data from EK-i/EW-i scale.
+
+    Data Sheet: https://www.aandd.co.jp/adhome/pdf/manual/balance/ekew-i.pdf
+    """
+
+    def __init__(self):
+        super(EkEwIDriver, self).__init__()
+        port = rospy.get_param('~port', '/dev/ttyUSB0')
+        timeout = rospy.get_param('~timeout', None)
+        rospy.loginfo('port=%s', port)
+        # EK-i/EW-i series default settings
+        self.ser = serial.Serial(
+            port, baudrate=2400, bytesize=7, parity=serial.PARITY_EVEN,
+            timeout=timeout, writeTimeout=timeout)
+        self.pub = rospy.Publisher('~output', WeightStamped, queue_size=1)
+        rate = rospy.get_param('~rate', 10)
+        self.read_timer = rospy.Timer(rospy.Duration(1. / rate),
+                                      self._read_timer_cb)
+
+    def _read_timer_cb(self, event):
+        if (self.pub.get_num_connections() == 0):
+            return
+
+        try:
+            self.ser.write('Q\r\n')
+        except SerialTimeoutException:
+            rospy.logerr('Serial write timeout')
+            rospy.signal_shutdown('Serial write timeout')
+            return
+        data = self.ser.read(17)
+        stamp = rospy.Time.now()
+        if len(data) != 17:
+            rospy.logerr('Serial read timeout')
+            rospy.signal_shutdown('Serial read timeout')
```

Comment:
Default value of `timeout` is `None`, so this shutdown never happens in default.
Anyway, I'll add launch file.